packages/components/radio/src/radio.vue

```diff
@@ -0,0 +1,98 @@
+<script setup lang="ts" name="ORadio">
+import { radioEmits, radioProps } from './radio'
+import { useRadio } from './use-radio'
+
+const props = defineProps(radioProps)
+const emit = defineEmits(radioEmits)
+
+const { radioRef, radioGroup, isFocus, size, isDisabled, modelValue } = useRadio(props, emit)
+const isBorder = computed(() => props.border)
+const isChecked = computed(() => modelValue.value === props.label)
+const fillColor = computed(() => `${props.labelColor || props.fill}Base`)
+const textColor = computed(() => props.notTextColor ? `${props.labelColor}Base` : `${props.textColor || props.fill}Base`)
+const labelColor = computed(() => `${props.labelColor ? `${props.labelColor}Base` : 'baseText'}`)
+
+function handleChange() {
+  nextTick(() => emit('change', modelValue.value))
+}
+</script>
+
+<template>
+  <label
+    class="o-radio-check-base"
+    :class="[
+      isDisabled && `o-disabled after:cursor-not-allowed after:bg-${fillColor}`,
+      `o-radio-${size}`,
+      isBorder && `o-radio-check-disabled ${isChecked ? `border-${fillColor}` : 'border-baseBorder'}`,
+      isChecked && `o-radio-check-checked`,
+    ]"
+  >
+    <!-- todo: delete this span -->
+    <!-- hack: class such as 'o-radio-${size} c-${textColor}' does not take effect, i must be set 'o-radio-xs o-radio-sm o-radio-md o-radio-lg etc' to make it work. -->
```

Comment:
You can set `o-radio-${size}` , `c-${textColor}` in uno's safeList.

Reply:
Thank you for your guidance. I've finished it.